acsoo/addons.py

```diff
@@ -85,3 +103,57 @@ def addons_list_depends(ctx, exclude):
 
 
 addons.add_command(addons_list_depends, 'list-depends')
+
+
+def do_makepot(config, database, odoo_bin, addons_dirs, odoo_config):
+    cmd = [
+        odoo_bin,
+        'shell',
+        '-d', database,
+        '--log-level=info',
+        '--no-xmlrpc',
+    ]
+    if odoo_config:
+        cmd.extend([
+            '-c', odoo_config
+        ])
+    script_dir = os.path.dirname(__file__)
+    file_path = os.path.join(script_dir, 'make_pot_script')
+    proc = subprocess.Popen(
+        cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT)
+    with open(file_path) as file:
+        cmd = file.read()
+    installable_addons = get_installable_addons(addons_dirs).items()
+    for addon, addon_dir in installable_addons:
+        kwargs = {
+            'module_name': addon,
+            'module_path': os.path.join(addon_dir[0], addon),
+        }
+        module_cmd = cmd % kwargs
+        proc.stdin.write(module_cmd)
+    proc.stdin.close()
+    checklog_config = config.get_default_map().get('checklog', {})
+    ignore = checklog_config.get('ignore', [])
+    echo = checklog_config.get('echo', None)
+    with tempinput(proc.stdout.read()) as tempfilename:
+        do_checklog(tempfilename, ignore, echo)
+    proc.terminate()
+
+
+@click.command()
+@click.option('--database')
+@click.option('--odoo-bin', 'odoo_bin', default='odoo',
+              type=click.Path(dir_okay=False, exists=True))
+@click.option('--addons-dir', 'addons_dirs', multiple=True,
+              type=click.Path(file_okay=False, exists=True),
+              help="Directory containing addons. Defaults to odoo/addons or "
+                   "odoo_addons if present. This option can be repeated.")
+@click.option('--odoo-config', 'odoo_config', default='odoo.cfg',
+              type=click.Path(dir_okay=False, exists=True))
+@click.pass_context
+def makepot(ctx, database, odoo_bin, addons_dirs, odoo_config):
+    do_makepot(ctx.obj['config'], database, odoo_bin, addons_dirs, odoo_config)
```

Comment:
can you move this to a separate module `addons_makepot.py` so all `addons` subcommand do not end up in the sale file.